### PR TITLE
Fix to allow classes with char members to deserialize

### PR DESCRIPTION
**Pull request by fenryka**

### CODE REVIEW

**Comment by fenryka** [comment]
The test file is small as I've other tests I'm working on that will go in as additional commits (pretty sure byte arrays just don't work at the moment either but am putting together some tests for that now)

File: core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt [hunk 1]
```diff
@@ -105,5 +106,21 @@ sealed class PropertySerializer(val name: String, val readMethod: Method, val re
             }
         }
     }
+
+    /**
+     * A property serializer for the AMQP char type, needed as a specialisation as the underlaying
```
Comment (rick-r3) [inline core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt]: underlaying -> underlying?

File: core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt [hunk 2]
```diff
@@ -105,5 +106,21 @@ sealed class PropertySerializer(val name: String, val readMethod: Method, val re
...
+     * value of the character is stored in numeric ASCII form and on deserialisation requires explicit
```
Comment (rick-r3) [inline core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt]: Really ASCII?  Or you mean UTF-16 16 bit value?

File: core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt [hunk 3]
```diff
@@ -105,5 +106,21 @@ sealed class PropertySerializer(val name: String, val readMethod: Method, val re
...
+     * casting back to a char otherwise it's treated as an Integer and TypeMismatchs occur
+     */
+    class AMQPCharPropertySerializer(name: String, readMethod: Method) :
+            PropertySerializer(name, readMethod, Char::class.java) {
+        override fun writeClassInfo(output: SerializationOutput) {}
+
+        override fun readProperty(obj: Any?, schema: Schema, input: DeserializationInput) = (obj as Int).toChar()
+
+        override fun writeProperty(obj: Any?, data: Data, output: SerializationOutput) {
+            data.putObject(readMethod.invoke(obj))
```
Comment (rick-r3) [inline core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt]: `data.putChar()`?

File: core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt [hunk 4]
```diff
@@ -55,8 +55,10 @@ sealed class PropertySerializer(val name: String, val readMethod: Method, val re
     companion object {
         fun make(name: String, readMethod: Method, resolvedType: Type, factory: SerializerFactory): PropertySerializer {
             if (SerializerFactory.isPrimitive(resolvedType)) {
-                // This is a little inefficient for performance since it does a runtime check of type.  We could do build time check with lots of subclasses here.
-                return AMQPPrimitivePropertySerializer(name, readMethod, resolvedType)
+                return when(resolvedType) {
+                    Char::class.java, Character::class.java -> AMQPCharPropertySerializer(name, readMethod)
```
Comment (rick-r3) [inline core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt]: Do we need to test character arrays?

File: core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt (hunk 2 above)
Comment (fenryka) [inline core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt]: well UTF16 with all of the  top bits set to zero so it's readable as ASCII

File: core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt (hunk 1 above)
Comment (fenryka) [inline core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt]: done

File: core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt (hunk 4 above)
Comment (fenryka) [inline core/src/main/kotlin/net/corda/core/serialization/amqp/PropertySerializer.kt]: I think they should work *if* arrays work... I've got a whole bunch of other tests / fixes coming so I'll add it to that commit if that's OK

**Pull request merged**